### ISSUE TITLE
Corner case in remediation template_BASH_service_disabled

### ISSUE THREAD

**Issue by OnceUponALoop** [OpenSCAP/scap-security-guide#2632]
#### Description of problem:
If a service is masked and enabled then `systemctl disable` will not actually disable it since the masking takes precedence.

#### SCAP Security Guide Version:
Git master 02/27 (bb897175d0695dd6b6ec434e727f66fb6eb13eea)

#### Operating System Version:
RHEL 7.4

#### Steps to Reproduce:
- Enable service (ex. kdump)
  `systemctl enable kdump.service`
- Mask the service
  `systemctl mask kdump.service`
- Remediate
- Verify results against oval 

#### Actual Results:
Check fails
```
[root@rh-test build]# oscap oval eval --id oval:ssg-service_kdump_disabled:def:1 ssg-rhel7-ds.xml
Definition oval:ssg-service_kdump_disabled:def:1: false
Evaluation done.
```

#### Expected Results:
Remediation should have fixed it

#### Addition Information/Debugging Steps:
It's a corner case and will be resolved after a reboot but the fix is easy enough.

The remediation script simply calls `systemctl disable %SERVICENAME%` which doesn't do anything since the service is masked. 

**Behavior if service is enabled but not masked**
```
# Get Status
[root@rh-test build]# systemctl status kdump.service
● kdump.service - Crash recovery kernel arming
   Loaded: loaded (/usr/lib/systemd/system/kdump.service; enabled; vendor preset: enabled)
   Active: inactive (dead) since Tue 2018-02-27 21:33:52 CST; 2h 29min ago
 Main PID: 22024 (code=exited, status=0/SUCCESS)

# Check multi-user.target.wants
[root@rh-test build]# ls -l /etc/systemd/system/multi-user.target.wants/ | grep kdump
lrwxrwxrwx  1 root root 37 Feb 27 23:51 kdump.service -> /usr/lib/systemd/system/kdump.service

# Disable (mimic remediation)
[root@rh-test build]# systemctl disable kdump.service
Removed symlink /etc/systemd/system/multi-user.target.wants/kdump.service.

# Check multi-user.target.wants
[root@rh-test build]# ls -l /etc/systemd/system/multi-user.target.wants/ | grep kdump
[root@rh-test build]#
```

**Behavior if service is enabled and masked**
```
# Get Status - note masked
[root@rh-test build]# systemctl status kdump.service
● kdump.service
   Loaded: masked (/dev/null; bad)
   Active: inactive (dead) since Tue 2018-02-27 21:33:52 CST; 2h 33min ago
 Main PID: 22024 (code=exited, status=0/SUCCESS)

# Check multi-user.target.wants
[root@rh-test build]# ls -l /etc/systemd/system/multi-user.target.wants/ | grep kdump
lrwxrwxrwx  1 root root 37 Feb 28 00:07 kdump.service -> /usr/lib/systemd/system/kdump.service

# Disable (mimic remediation) - note no "Removed symlink ..." message
[root@rh-test build]# systemctl disable kdump.service
[root@rh-test build]#

# Check multi-user.target.wants
[root@rh-test build]# ls -l /etc/systemd/system/multi-user.target.wants/ | grep kdump
lrwxrwxrwx  1 root root 37 Feb 28 00:07 kdump.service -> /usr/lib/systemd/system/kdump.service
```

### Proposed Solution ###

[Edit]
Initially i proposed the script below to capture the masked status and alter behavior accordingly but since then I've looked into this some more and it would require updating the `shared/bash_remediation_functions/service_command.sh` or creating a different one to handle the "is-enabled" option. 

I was hesitant to alter the service_command.sh since that would make it systemd specific. 

So that leaves me with a few questions for yall
- Is this solution acceptable? 
- How to best implement it (new bash_remediation_function)?

<s>Change the remediation script `rhel7/templates/template_BASH_service_disabled` to check if masked. If masked, unmask, disable, then re-mask.
``` bash
# Get service status
serviceStatus="$(service_command is-enabled %SERVICENAME%)"

# Check if masked
if [[ "${serviceStatus}" == "masked" || "${serviceStatus}" == "masked-runtime" ]]; then 
    # Unmask, disable, re-apply mask
    service_command unmask  %SERVICENAME%
    service_command disable %SERVICENAME%
    service_command mask    %SERVICENAME%
else
    # Not masked, just disable it
    service_command disable %SERVICENAME%
fi
```
</s>

**Comment by redhatrises** [comment]
Correct me if I am wrong, but isn't a masked service linked to `/dev/null` rendering the service completely unstartable? If so, I wonder if the OVAL check should be extended to check for masked services instead.

**Comment by OnceUponALoop** [comment]
@redhatrises The masking behavior surprised me, guess I never really looked into the semantics of it all until now.

1. Masking an enabled service doesn't remove symlinks from *target.wants even after a reboot.
2. Masking a running service doesn't stop it but will prevent it from being started on reboot.
  Verified against sshd since kdump doesn't spawn a process.
3. Once a service is masked systemctl ignores control commands except for unmask.

If we update the oval to check the mask status (linked to `/dev/null`) we'll still need to consider the case where a service was masked after it started. 

That's just the checks, remediation will still have the same issues 😞

**Comment by redhatrises** [comment]
Expectation is that the system is rebooted hence `reboot="true"` in the remediations as well as stated in the guild.

So to keep in line with everything else, I would just extend the OVAL check.

**Comment by redhatrises** [comment]
@OnceUponALoop I believe that this has been fixed in the latest master. Can you confirm?

**Comment by redhatrises** [comment]
Should be fixed by #2698. Please reopen if it is not fixed.
